### ISSUE TITLE
Is Adaptive composite blend‘s bug?

### ISSUE THREAD

**Issue by nvnv-sunny** [MiSTer-devel/Genesis_MiSTer#125]
![20200505071543](https://user-images.githubusercontent.com/62794981/81022542-71993900-8ea0-11ea-84bc-fe0a4c26ec90.jpg)

**Comment by sorgelig** [comment]
no. it's not a bug. Only transparency alternating pixels can be blended in adaptive mode. 
Embedded into tiles dithering patterns cannot be blended.

**Comment by MJY71** [comment]
I was curious about that as well. There appears to be inconsistent behavior with blending alternating pixels. The example here is the shadows on the ground from the mountains in Lunar: The Silver Star.
I know this screenshot is from MegaCD, but it's the same filter, right?
![20200503_072534-screen-enlarged](https://user-images.githubusercontent.com/18260997/81117422-ad59f000-8eec-11ea-9078-25fb79ee934e.png)

**Comment by sorgelig** [comment]
There is no magic. It works only when overlaid foreground has alternating transparent pixels. On picture above some tiles are on foreground with transparency, some tiles are on background with non-transparent dithering. It cannot be fixed. Originally the whole screen should be blended anyway.

**Comment by MJY71** [comment]
Good point. Thank you for taking the time to respond.

**Comment by nvnv-sunny** [comment]
Whether the background pixels can be mixed with transparent layer again  only for background pixels at the bottom of the transparency layer

**Comment by sorgelig** [comment]
transparency on background layer is not used because it may give opposite artifacts when invisible(covered) tiles will add transparency on foreground layer.

At the same time if someone will come with a better code doing adaptive blend, it will be used.

**Comment by nvnv-sunny** [comment]
Oh ,i see. Thanks for your reply.

**Comment by Daddu3** [comment]
> transparency on background layer is not used because it may give opposite artifacts when invisible(covered) tiles will add transparency on foreground layer.
> 
> At the same time if someone will come with a better code doing adaptive blend, it will be used.

I found this beta core developed for Genesis a long while back, which looks to implement a more improved blending - however, the dev for this beta has been MIA for some while.

https://misterfpga.org/viewtopic.php?p=41511#p41511


Some observations I made via some Sonic-games and some:

In Sonic 1, the Smart Blending via "Enabled"-setting (instead of "Composite",  or "Transparent"), has both the transparency effects for Shield-pickup and the waterfalls render properly together while keeping the sharp pixels. Dithering's looking pretty seamless too in Ecco The Dolphin

However, in Sonic 2 the blending seems to bit hit or miss - the palm trees blend correctly but the waterfall and bushes on Emerald Hill Zone doesn't seem to work. Meanwhile in Sonic Spinball the dithering pattern for the toxic slime is pretty glitched.

At least this seems to tell me achieving a more universal blending could be possible on this core.
